README.md

```diff
@@ -2,53 +2,179 @@
+checked into version control, it is helpful to know that the real manifest is
+named `addon-manifest.json` and will be called "the manifest" from this point
+forward.
+
+## Basic auth
+
+### Heroku docs
+
+* https://devcenter.heroku.com/articles/building-an-add-on#basic-authentication
+
+### Related code
+
+Basic auth is implemented via the [`HttpBasicAuth`
+concern](app/controllers/concerns/http_basic_auth.rb). This concern is included
+in [`ApplicationController`](app/controllers/application_controller.rb), which
+all other controllers inherit from, because the Add-on Partner API sends basic
+auth credentials with all requests. Basic auth is selectively skipped for the
+single sign-on views, which are accessed by Heroku customers who are using the
+add-on rather than the Add-on Partner API.
+
+The username and password for basic auth are being accessed via environment
+variables whose values match the `slug` and `password` fields in the manifest.
+
+## The provisioning request
+
+### Heroku docs
+
+* https://devcenter.heroku.com/articles/building-an-add-on#the-provisioning-request
+* https://devcenter.heroku.com/articles/getting-started-with-asynchronous-provisioning#implementation
+* https://devcenter.heroku.com/articles/add-on-partner-api-reference#add-on-action-create-provision
+* https://devcenter.heroku.com/articles/platform-api-reference#add-on-action-provision
+
+### Related code
+
+The Sudo Sandwich add-on is provisioned synchronously or asynchronously
+depending on the plan that is selected. This somewhat artificial plan slug
+distinction is just to demonstrate the process of provisioning plans
+asynchronously and synchronously.
+
+For both types of provisioning, the `base_url` key in the add-on manifest
+specifies a path of `/heroku/resources`, which is routed to the
+[`Heroku::ResourcesController`](app/controllers/heroku/resources_controller.rb)
+`#create` method in this codebase. This is the endpoint that Heroku hits when a
+customer creates an add-on resource.
+
+#### Synchronous provisioning
+
+The controller looks for the `plan` param, and if it matches
+`Sandwich::BASE_PLAN` it returns a status code of 200, which tells Heroku that
+the add-on is being provisioned synchronously, and a message that tells
+the add-on customer that their add-on is available immediately.
+
+#### Asynchronous provisioning
+
+For all other plans, a status code of 202 is returned, which tells
+Heroku that the add-on is being provisioned asynchronously. For plans that are
+being provisioned asynchronously, an `access_token` is required. Retrieving an
+`access_token` is covered in the "Grant code exchange" section.
+
+Once the `access_token` is available, a plan is marked as `provisioned` by
+running the [`ProvisionPlanJob`](app/jobs/provision_plan_job.rb). This
+background job calls [`PlanProvisioner`](app/services/plan_provisioner.rb),
+which sends a POST request to the Add-on Partner API. This request includes the
```

Comment:
"Add-on Partner API" should be "Platform API for Partners"

Reply:
updated at 6b40c136237196b5227e90175420c24e5d953fe5